ios/ios/Scene/SignUp/SignUpViewController.swift

```diff
@@ -0,0 +1,269 @@
+    codeTextField.layer.cornerRadius = Constant.textFieldCornerRadius
+    codeTextField.layer.borderWidth = Constant.textFieldBorderWidth
+    codeTextField.layer.borderColor = UIColor.black.cgColor
+  }
+
+  private func showTimerLabel() {
+    timerLabel.translatesAutoresizingMaskIntoConstraints = false
+    view.addSubview(timerLabel)
+    timerLabel.topAnchor.constraint(equalTo: getCodeButton.bottomAnchor, constant: 10).isActive = true
+    timerLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -5).isActive = true
+    timerLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 5).isActive = true
+    timerLabel.heightAnchor.constraint(equalToConstant: 50).isActive = true
+    timerLabel.textAlignment = .center
+    timerLabel.numberOfLines = 0
+    timerLabel.textColor = .darkGray
+    var secondCount = 20
+    Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true) { timer in
+      secondCount -= 1
+      self.timerLabel.text = "Можете запросить код повторно через: \(secondCount) секунд"
+      if secondCount == 0 {
+        self.timerLabel.text = ""
+        timer.invalidate()
+        self.enableButton(isEnabled: true)
+      }
+    }
+  }
+
+  // MARK: - private textField methods
+  @objc private func nameTextFieldFilled(_ sender: UITextField) {
+    defer { enableButton(isEnabled: nameIsNotEmty && surnamtIsNotEmpty && phoneIsFill) }
+    guard
+      let count = sender.text?.count,
+      count >= 3
+    else {
+      sender.text = "  "
+      nameIsNotEmty = false
+      return
+    }
+    nameIsNotEmty = true
+  }
+
+  @objc private func surnameTextFieldFilled(_ sender: UITextField) {
+    defer { enableButton(isEnabled: nameIsNotEmty && surnamtIsNotEmpty && phoneIsFill) }
+    guard
+      let count = sender.text?.count,
+      count >= 3
+    else {
+      sender.text = "  "
+      surnamtIsNotEmpty = false
+      return
+    }
+    surnamtIsNotEmpty = true
+  }
+
+  @objc private func phoneTextFieldFilled(_ sender: UITextField) {
+    defer { enableButton(isEnabled: nameIsNotEmty && surnamtIsNotEmpty && phoneIsFill) }
+    guard
+      let count = sender.text?.count,
+      count >= Constant.phoneLength
+    else {
+      phoneIsFill = false
+      return
+    }
+    phoneIsFill = true
+  }
+
+  // MARK: - private buttons methods
+  @objc private func getCodeTapped(_ sender: UIButton) {
+    onRequestedComplete()
+        guard
+          let name = nameTextField.text,
+          let lastName = surnameTextField.text,
+          let phone = phoneTextField.text
+        else {
+          return
+        }
+        interactor?.signUpDidRequestVerificationCode(
+          self,
+          forSignUp: SignUp(
+          firstName: name, lastName: lastName, phone: phone)
+        )
+        enableButton(isEnabled: false)
+  }
+
+  @objc private func registerTapped(_ sender: UIButton) {
+    fatalError("Not implemented yet")
+  }
+
+  // MARK: - SignUpView methods
+  func onRequestedComplete() {
```

Comment:
It'd better be `onRequestCompleted` or `onCompleteRequested` (which one was meant?)